feat: add C ndarray interface and refactor implementation for `stats/base/dnanmskrange`

gururaj1512 opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

- adds C ndarray interface for `stats/base/dnanmskrange`
- refactors JavaScript implementation
- refactors tests, docs, benchmarks and examples.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add C ndarray interface and refactor implementation for `stats/base/dnanmskrange`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
docs: update README

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
````diff
@@ -188,6 +178,143 @@ console.log( v );
+double stdlib_strided_dnanmskrange( const CBLAS_INT N, const double *X, const CBLAS_INT strideX, const uint8_t *Mask, const CBLAS_INT strideMask );
+```
+
+#### stdlib_strided_dnanmskrange_ndarray( N, \*X, strideX, offsetX, \*Mask, strideMask, offsetMask )
````
How to deal with lint error here? `min header len 80`

kgryte replied:
This should now be resolved.

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
```diff
@@ -153,31 +146,28 @@ var v = dnanmskrange.ndarray( N, x, 2, 1, mask, 2, 1 );
+var uniform = require( '@stdlib/random/base/uniform' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var uint8Array = require( '@stdlib/random/array/bernoulli' );
```
Here both `@stdlib/random/array/bernoulli` and `@stdlib/random/base/bernoulli` are being used...!
What should be the namings? Is uint8Array okay?

kgryte committed:
docs: clean-up

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: clean-up example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/benchmark/benchmark.js:
```diff
@@ -21,17 +21,38 @@
-var randu = require( '@stdlib/random/base/randu' );
+var uniform = require( '@stdlib/random/base/uniform' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var uint8Array = require( '@stdlib/random/array/bernoulli' );
```
Use the same approach as in the examples. Namely, just use the `factory` method from `base/bernoulli`. Applies here and to the other benchmarks.

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
````diff
@@ -188,6 +174,143 @@ console.log( v );
+
+```c
+const double x[] = { 1.0, -2.0, 4.0, 2.0, 0.0/0.0 };
+const uint8_t mask[] = { 0, 0, 1, 0, 0 };
````
```suggestion
const uint8_t mask[] = { 0, 0, 0, 0, 1 };
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
```diff
@@ -188,6 +174,143 @@ console.log( v );
+#include <stdint.h>
+
+const double x[] = { 1.0, -2.0, 4.0, 2.0, 0.0/0.0 };
+const uint8_t mask[] = { 0, 0, 1, 0, 0 };
```
```suggestion
const uint8_t mask[] = { 0, 0, 0, 0, 1 };
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
```diff
@@ -188,6 +174,143 @@ console.log( v );
+#include <stdint.h>
+
+const double x[] = { 1.0, -2.0, 4.0, 2.0, 0.0/0.0 };
+const uint8_t mask[] = { 0, 0, 0, 0, 1 };
```
```suggestion
const uint8_t mask[] = { 0, 0, 1, 0, 0 };
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/README.md:
````diff
@@ -188,6 +174,143 @@ console.log( v );
+
+```c
+const double x[] = { 1.0, -2.0, 4.0, 2.0, 0.0/0.0 };
+const uint8_t mask[] = { 0, 0, 0, 0, 1 };
````
```suggestion
const uint8_t mask[] = { 0, 0, 1, 0, 0 };
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dnanmskrange/src/main.c:
```diff
@@ -30,27 +31,37 @@
+}
+
+/**
+* Computes the range of a double-precision floating-point strided array according to a mask, ignoring `NaN` values and alternative indexing semantics.
```
```suggestion
* Computes the range of a double-precision floating-point strided array according to a mask, ignoring `NaN` values and using alternative indexing semantics.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (changes requested):
Apart from the benchmarks, this is looking good.

gururaj1512 committed:
bench: update benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
Update dnanmskrange.js

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 committed:
Update README.md

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

kgryte reviewed (approved):
LGTM
